app/store/location-store.coffee

```diff
@@ -46,6 +46,12 @@ class LocationStore extends Store
     @status = @STATUS_FOUND_LOCATION
     @emitChange()
 
+  storeLocationWithoutEmit: (location) ->
+    @lat = location.lat
+    @lon = location.lon
+    @heading = if location.heading then location.heading else @heading
+    @status = @STATUS_FOUND_LOCATION
+
```

Comment:
Would be good to comment here shortly why we sometimes want to emit and sometimes not.

This is pretty much the same as storeLocation, but heading is missing in the old function. If intentional, would be good to insert a comment on that.

Also, could the storeLocation simply call storeLocationWithoutEmit and then emit, to remove code duplication?